e2e-tests/playwright/support/pages/adoption-Insights.ts

```diff
@@ -0,0 +1,146 @@
+  
+      if (match) {
+        return parseInt(match[0], 10);
+      }
+  
+      return null; // or 0 if you'd prefer a default
+    } catch (error) {
+      console.error('Error getting count from panel:', error);
+      return null;
+    }
+  }
+  
+  async getVisibleFirstRowText(panel: Locator): Promise<string[]> {
+    const firstRow = panel.locator('table.v5-MuiTable-root tbody tr').first();
+  
+    if (await firstRow.isVisible()) {
+      const cells = firstRow.locator('td');
+      const cellCount = await cells.count();
+      const texts: string[] = [];
+  
+      for (let i = 0; i < cellCount; i++) {
+        const cellText = await cells.nth(i).textContent();
+        texts.push(cellText?.trim() ?? '');
+      }
+      
+      // Return first and last elements
+      return [texts[0], texts[texts.length - 1]];
+    }
+    return [];
+  }
+
+  async populateMissingPanelData(
+    page: Page,
+    uiHelper: any,
+    templatesFirstLast: string[],
+    catalogEntitiesFirstLast: string[],
+    techdocsFirstLast: string[]
+  ): Promise<void> {
+    if (templatesFirstLast.length === 0) {
+      // Run a template
+      const inputText = 'reallyUniqueName';
+      await page.getByRole('link', { name: 'Self-service' }).click();
+      const chooseButton = page.getByRole('button', { name: 'Choose' });
+      await chooseButton.last().click();
+      await uiHelper.fillTextInputByLabel('Organization', inputText);
+      await uiHelper.fillTextInputByLabel('Repository', inputText);
+      await uiHelper.clickButton("Next");
+      await uiHelper.fillTextInputByLabel('Image Builder', inputText);
+      await uiHelper.fillTextInputByLabel('Image URL', inputText);
+      await uiHelper.fillTextInputByLabel('Namespace', inputText);
+      await uiHelper.fillTextInputByLabel('Port', '8080');
+      await uiHelper.clickButton('Review');
+      await uiHelper.clickButton('Create');
+      await page.getByText("Run of Create a tekton CI").waitFor({ state: "visible" });
+    }
+  
+    if (catalogEntitiesFirstLast.length === 0) {
+      // Visit a catalog entity
+      await uiHelper.clickLink("Catalog");
+      await uiHelper.clickLink('Red Hat Developer Hub');
+      await expect(page.getByText('Red Hat Developer Hub')).toBeVisible();
+    }
+  
+    if (techdocsFirstLast.length === 0) {
+      // Visit docs
+      await uiHelper.clickLink('docs');
+      await uiHelper.clickLink('Red Hat Developer Hub');
+    }
+  }
+
+  async clickAndVerifyText(
+    firstEntry: Locator,
+    expectedText: string
+  ): Promise<void> {
+    const [newpage] = await Promise.all([
+      this.page.waitForEvent('popup'),
+      firstEntry.locator('a').click(),
+    ]);
+      // Wait for the expected API call to succeed
+    await this.waitUntilApiCallSucceeds(newpage, '/api/adoption-insights/events');
+
+    await newpage.getByText(expectedText).first().waitFor({ state: 'visible' });
+  }
+
+  async waitUntilApiCallSucceeds(page: Page, urlPart: string): Promise<void> {
```

Comment:
might be handy to put `'/api/adoption-insights/events'` as the default endpoint matcher